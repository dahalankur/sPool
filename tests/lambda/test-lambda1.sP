# lambdas with zero, one or more formals
lambda int (): return 0;
lambda int (int x): return x;
lambda int (int x, bool y, string z): return x;
