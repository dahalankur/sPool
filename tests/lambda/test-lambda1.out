PROGRAM {[ 
    LAMBDA(INTTY, FORMALS(), [ 
        RETURN(INT(0))]) 
    LAMBDA(INTTY, FORMALS((INTTY, x)), [ 
        RETURN(VAR(x))]) 
    LAMBDA(INTTY, FORMALS((INTTY, x), (BOOLTY, y), (STRINGTY, z)), [ 
        RETURN(VAR(x))])]}
